Mark AuxiliaryProcessProxy::fromConnection as nullable

adetaylor opened the pull request:
#### e8002401c67d6d51017851e9d9b2004ab82d671f
<pre>
Mark AuxiliaryProcessProxy::fromConnection as nullable
<a href="https://bugs.webkit.org/show_bug.cgi?id=302643">https://bugs.webkit.org/show_bug.cgi?id=302643</a>
<a href="https://rdar.apple.com/164887710">rdar://164887710</a>

Reviewed by Geoffrey Garen.

This marks this API as returning a value which may potentially be null. This is
necessary for Swift/C++ interoperability to work properly; for our current
prototypes of a CoreIPC message receiver in Swift it&apos;s the only such API which
needs to be so annotated.

This suppresses a related warning in cmake builds which is already suppressed
in Xcode builds.

Canonical link: <a href="https://commits.webkit.org/303248@main">https://commits.webkit.org/303248@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/19dc91b32b724e31f83923aad9febea18abd41c2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131618 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3950 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42583 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139127 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83414 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a2e395bb-2434-4419-a933-942c812f8f98) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3981 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3791 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100481 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68064 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134564 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2845 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117807 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81289 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2760 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/517 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82318 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/123643 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111386 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35933 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141772 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/130072 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3694 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36449 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108856 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3742 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3275 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109099 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27680 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2799 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114136 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56899 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3755 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32537 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/163089 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3582 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67166 "Built successfully") | [⏳ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3837 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3685 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/cmake/WebKitCompilerFlags.cmake:
```diff
@@ -192,7 +192,8 @@ if (COMPILER_IS_GCC_OR_CLANG)
                                          -Wno-misleading-indentation
-                                         -Wno-psabi)
+                                         -Wno-psabi
+                                         -Wno-nullability-completeness)
```
Seems like a pretty big change for a commit named "Mark AuxiliaryProcessProxy::fromConnection as nullable."

geoffreygaren replied:
FWIW, this was discussed in [WebKit Slack](https://webkit.slack.com/archives/CU64U6FDW/p1761239537764009).

Basically, it's not practical to mark something nullable unless you first disable `-Wnullability-completeness` (and relatedly, `-Wnullability-completeness` has no meaning until you first mark something nullable).

We must have missed this when we made [the same change for Apple Ports](https://commits.webkit.org/302237@main), probably because we did not have Swift building with CMake yet.

geoffreygaren replied:
But yeah, @adetaylor probably a good idea to mention in the commit message that this change also flips that flag for the first time in CMake builds.

adetaylor replied:
Thanks - yep, the _commit_ message already did state that, but I hadn't updated the PR description to match the commit description. Now done.

adetaylor committed:
Mark AuxiliaryProcessProxy::fromConnection as nullable
https://bugs.webkit.org/show_bug.cgi?id=302643
rdar://164887710

Reviewed by Geoffrey Garen.

This marks this API as returning a value which may potentially be null. This is
necessary for Swift/C++ interoperability to work properly; for our current
prototypes of a CoreIPC message receiver in Swift it's the only such API which
needs to be so annotated.

This suppresses a related warning in cmake builds which is already suppressed
in Xcode builds.

Canonical link: https://commits.webkit.org/303248@main
